conda/core/subdir_data.py

```diff
@@ -170,16 +154,27 @@ def __init__(self, channel, repodata_fn=REPODATA_FN):
         assert channel.subdir
         if channel.package_filename:
             parts = channel.dump()
-            del parts['package_filename']
+            del parts["package_filename"]
             channel = Channel(**parts)
         self.channel = channel
-        self.url_w_subdir = self.channel.url(with_credentials=False)
-        self.url_w_credentials = self.channel.url(with_credentials=True)
+        # disallow None (typing)
+        self.url_w_subdir = self.channel.url(with_credentials=False) or ""
+        self.url_w_credentials = self.channel.url(with_credentials=True) or ""
         # whether or not to try using the new, trimmed-down repodata
         self.repodata_fn = repodata_fn
         self._loaded = False
         self._key_mgr = None
 
+    @property
+    def _repo(self) -> RepoInterface:
+        # XXX one for `current_repodata.json` and one for `repodata.json`
```

Comment:
The class changes its url when retrying repodata.json, when current_repodata.json fails to load.

Reply:
I think I don't hate this anymore. This object is not long-lived but that's fine.